core/jazz_environment-event-handler/index.js

```diff
@@ -597,6 +601,265 @@ function triggerBuildJob(payload, serviceDetails, configData) {
   });
 }
 
+function createSafe(environmentPayload, service_id, configData, authToken) {
+  return new Promise((resolve, reject) => {
+    var updatePayload = {};
+    const safeName = environmentPayload.service + '_' + environmentPayload.domain + '_' + environmentPayload.logical_id;
+    updatePayload.name = safeName;
+    updatePayload.owner = configData.SERVICE_USER;
+    updatePayload.description = "create safe for jazz tvault service: " + safeName;
+
+    var svcPayload = {
+      uri: configData.BASE_API_URL + '/' + environmentPayload.logical_id + configData.TVAULT.API
+        + `?domain=${environmentPayload.domain}&service=${environmentPayload.service}&environment=${environmentPayload.logical_id}`,
+      method: "POST",
+      headers: {
+        "Authorization": authToken,
+        "Content-Type": "application/json",
+        "Jazz-Service-ID": service_id
+      },
+      json: updatePayload,
+      rejectUnauthorized: false
+    };
+    request(svcPayload, function (error, response, body) {
+      if (response.statusCode && response.statusCode === 200) {
+        const timestamp = new Date().toISOString();
+        const vaultLink = configData.TVAULT.HOSTNAME + "/#!/admin";
+        environmentPayload['safe_details'] = { "name": safeName, "ts": timestamp, "link": vaultLink };
+        safeDetails = environmentPayload['safe_details'];
+        return resolve(environmentPayload);
```

Comment:
I can see some additional datas being injecting and resolving that data. But I couldn't see the use of that anywhere. Whats the purpose of that?

Reply:
As discussed, added to env-payload to update env metadata and changed accordingly.

Reply:
And Used hostname for deep link.